### DISCUSSION TITLE
Panic when analyzing keyval map

### DISCUSSION THREAD

**Discussion by elskndd** [keyval-dev/opentelemetry-go-instrumentation#46] [Q&A] [answered]
Hi,

I am trying this auto instrumentation tool. But I got panic like following.
![image](https://user-images.githubusercontent.com/54382019/206902645-424124cc-dfe1-4bec-9c25-bfb8f30b5852.png)
I went through the code and found there is a check here.
[https://github.com/keyval-dev/opentelemetry-go-instrumentation/blob/master/pkg/process/analyze.go#L72](url)
![image](https://user-images.githubusercontent.com/54382019/206902842-2676b915-358d-4cad-836d-0b596322a449.png)

I don't understand why the permission should be "**rwx**" here. And I cheched /proc/pid/maps and found there is no map with permission "**rwx**".

Would anyone help me here? Thanks!

**Comment by edeNFed** [comment, marked as answer]
Did you invoke the application using the launcher process?
Look at this file for an example:
https://github.com/keyval-dev/opentelemetry-go-instrumentation/blob/c4799f7d5fbedcd7533daf0e4103449c6ed156e5/docs/getting-started/emojivoto-instrumented.yaml#L44-L46

**Reply by elskndd** [reply]
@edeNFed hi, thanks for your help. I tried using the launcher and it worked.
